llvm/include/llvm/IR/Value.h

```diff
@@ -518,17 +509,8 @@ class Value {
 
   /// This method should only be used by the Use class.
   void addUse(Use &U) {
-    if (hasUseList())
-      U.addToList(Uses.List);
-    else
-      U.addToList(Uses.Count);
-  }
-
-  void removeUse(Use &U) {
-    if (hasUseList())
-      U.removeFromList(Uses.List);
-    else
-      U.removeFromList(Uses.Count);
+    if (UseList || hasUseList())
```

Comment:
Why the `UseList ||` here? Isn't `hasUseList()` sufficient? Or is this just an optimization?

Reply:
hasUseList is sufficient, not sure if this is really an optimization. I was going to remove this after to see if it makes a difference